src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/General/MaintenanceType.php

```diff
@@ -75,6 +116,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'locales' => $this->locales,
                     'hideTabs' => false,
                     'required' => true,
+                    'disabled' => true,
+                    'attr' => [
+                        'multistore_configuration_key' => 'PS_MAINTENANCE_TEXT',
+                        'disabled' => !$this->isAllShopContext && !$this->shopConfiguration->isOverridenByCurrentContext('PS_MAINTENANCE_TEXT'),
```

Comment:
Actually that's what I wanted to do but I get this error:

`FormConfigBuilder methods cannot be accessed anymore once the builder is turned into a FormConfigInterface instance.`

Reply:
Mmm, then maybe we shouldn't rely on the form extension then I actually had an alternative in mind

Instead of using an extension we create a service, for example `MultistoreConfigurationFormBuilder`, that is responsible for dynamically add all these checkboxes fields It doesn't even need to rely on FormEvents it would just have a method: `public function addMultistoreCheckboxes(FormBuilderInterface $builder): void` that would be called at the end of the `buildForm` method

Reply:
But before you do that one thing you can try, the extension currently relies on `FormEvents::PRE_SET_DATA` but I'm not sure it is necessary It's usually used when we want to dynamically build the form based on the request but that's not necessary here

So maybe you could add the extra fields directly using the `FormBuilderInterface $builder`

Reply:
Unfortunately it didn't work, seems like I'll have to go for the `MultistoreConfigurationFormBuilder ` service

Reply:
I found a way to do it inside the form extension, not sure if it's the best way though 🤔  But it does work